src/main/java/frc/robot/Vision/Limelight.java

```diff
@@ -0,0 +1,214 @@
+package frc.robot.Vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Vision.LimelightHelpers.LimelightResults;
+
+public class Limelight extends SubsystemBase {
+    private final NetworkTable table;
+
+    private final NetworkTableEntry tv;
+    private final NetworkTableEntry tx;
+    private final NetworkTableEntry ty;
+    private final NetworkTableEntry ta;
+    private final NetworkTableEntry ts;
+    private final NetworkTableEntry tl;
+
+    private final NetworkTableEntry tCornX;
+    private final NetworkTableEntry tCornY;
+
+    private final NetworkTableEntry ledMode;
+    private final NetworkTableEntry camMode;
+    private final NetworkTableEntry pipeline;
+    private final NetworkTableEntry stream;
+    private final NetworkTableEntry snapshot;
+
+    private final NetworkTableEntry tid;
+    private final NetworkTableEntry botpose;
+
+    private LimelightResults llresults = LimelightHelpers.getLatestResults("limelight");
```

Comment:
The reason your code works in `DrivebaseSubsysem.periodic` and not in this Limelight subsystem is right here. You're parsing your limelight data just once, on this line, when you construct the subsystem, and never again. You have methods below which refer to this one blip of data and never re-read it.

If the drivebase, you're actually re-reading and parsing your JSON limelight data every time. Which is correct.